addons/escoria-core/game/core-scripts/esc_controller.gd

```diff
@@ -105,20 +109,23 @@ func perform_inputevent_on_object(
 	
 	# If clicked object not in inventory, player walks towards it
 	if not obj.node is ESCPlayer and \
-			not escoria.inventory_manager.inventory_has(obj.global_id):
-		var context = _walk_towards_object(
-			obj, 
-			event.position, 
-			event.doubleclick
-		)
-		if context is GDScriptFunctionState:
-			context = yield(_walk_towards_object(
+		not escoria.inventory_manager.inventory_has(obj.global_id) and \
+		(not obj.events.has(escoria.action_manager.current_action) \
```

Comment:
Why don't you use the `event_flags` variable you introduced in line 99 here. It would simplify this if a lot.

Reply:
Done. Thanks!